eng/packages/http-client-csharp-mgmt/generator/TestProjects/Local/Mgmt-TypeSpec/src/Generated/FooResource.Serialization.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.ClientModel.Primitives;
+using System.Text.Json;
+using MgmtTypeSpec.Models;
+
+namespace MgmtTypeSpec
+{
+    /// <summary></summary>
+    public partial class FooResource : IJsonModel<FooData>
+    {
+        private static FooData s_dataDeserializationInstance;
+
+        private static FooData DataDeserializationInstance => s_dataDeserializationInstance = new FooData();
```

Comment:
could we update the type here to be `IJsonModel<FooData>` so that we could remove the cast expressions below?

Reply:
also I think this should be
```suggestion
        private static FooData DataDeserializationInstance => s_dataDeserializationInstance ??= new FooData();
```